core/src/main/java/org/apache/iceberg/ContentFileParser.java

```diff
@@ -147,7 +155,7 @@ public static ContentFile<?> fromJson(JsonNode jsonNode, Map<Integer, PartitionS
     int specId = JsonUtil.getInt(SPEC_ID, jsonNode);
     PartitionSpec spec = specsById.get(specId);
     Preconditions.checkArgument(spec != null, "Invalid partition specId: %s", specId);
-    FileContent fileContent = FileContent.valueOf(JsonUtil.getString(CONTENT, jsonNode));
+    FileContent fileContent = fileContentFromJson(JsonUtil.getString(CONTENT, jsonNode));
```

Comment:
Minor: fileContentFromString

I also wouldn't be opposed to introducing a FileContent.fromString API that does what fileContentFromString does below.